libcudacxx/include/cuda/std/detail/libcxx/include/__algorithm/for_each.h

```diff
@@ -22,7 +22,7 @@
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
 template <class _InputIterator, class _Function>
-_LIBCUDACXX_NODISCARD_EXT inline _LIBCUDACXX_HIDE_FROM_ABI _LIBCUDACXX_INLINE_VISIBILITY _CCCL_CONSTEXPR_CXX14 _Function
```

Comment:
question: just to make sure, nodiscard disappeared because we think the return value is rarely used?

Reply:
Yes, to clarify a bit more. 

libc++ started adding nodiscard to more functions. But those were marked as `_LIBCUDACXX_NODISCARD_EXT` as marking them is not standard mandated.

However, I believe we should not introduce `[[nodiscard]]` for functions that have a side effect. That only acts as a annoyance for users and does not serve values. That said, I plan on adding `[[nodiscard]]` to all other functions that do not have usable side effects